### ISSUE TITLE
rails g activeadmin_addons:install broken in rails 6.0.3.1

### ISSUE THREAD

**Issue by islador** [platanus/activeadmin_addons#321]
```
rails g activeadmin_addons:install
```
does not work on Rails 6.0.3.1 and returns the below error.

```
rails g activeadmin_addons:install
Running via Spring preloader in process 2872
Deprecation warning: Expected string default value for '--controller'; got false (boolean).
This will be rejected in the future unless you explicitly pass the options `check_default_type: false` or call `allow_incompatible_default_type!` in your code
You can silence deprecations warning by setting the environment variable THOR_SILENCE_DEPRECATION.
Could not find generator 'activeadmin_addons:install'. Maybe you meant "active_admin:install"?
```

I was able to work around this issue thanks to https://github.com/platanus/activeadmin_addons/blob/master/docs/install_generator.md.

**Comment by sleivav** [comment]
I tried replicating this and it worked correctly on a fresh rails 6.0.3.1 project, after adding and installing activeadmin and activeadmin_addons.

What version of those gems are you using?

**Comment by islador** [comment]
Sorry for the delay, been busy at work. 

rails (6.0.3.1)
activeadmin (2.7.0)
activeadmin_addons (1.7.1)

Full gemfile.lock:
```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (6.0.3.1)
      actionpack (= 6.0.3.1)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailbox (6.0.3.1)
      actionpack (= 6.0.3.1)
      activejob (= 6.0.3.1)
      activerecord (= 6.0.3.1)
      activestorage (= 6.0.3.1)
      activesupport (= 6.0.3.1)
      mail (>= 2.7.1)
    actionmailer (6.0.3.1)
      actionpack (= 6.0.3.1)
      actionview (= 6.0.3.1)
      activejob (= 6.0.3.1)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (6.0.3.1)
      actionview (= 6.0.3.1)
      activesupport (= 6.0.3.1)
      rack (~> 2.0, >= 2.0.8)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.2.0)
    actiontext (6.0.3.1)
      actionpack (= 6.0.3.1)
      activerecord (= 6.0.3.1)
      activestorage (= 6.0.3.1)
      activesupport (= 6.0.3.1)
      nokogiri (>= 1.8.5)
    actionview (6.0.3.1)
      activesupport (= 6.0.3.1)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    active_material (1.4.2)
    activeadmin (2.7.0)
      arbre (~> 1.2, >= 1.2.1)
      formtastic (~> 3.1)
      formtastic_i18n (~> 0.4)
      inherited_resources (~> 1.7)
      jquery-rails (~> 4.2)
      kaminari (~> 1.0, >= 1.0.1)
      railties (>= 5.2, < 6.1)
      ransack (~> 2.1, >= 2.1.1)
      sassc-rails (~> 2.1)
      sprockets (>= 3.0, < 4.1)
    activeadmin_addons (1.7.1)
      active_material
      railties
      require_all (~> 1.5)
      sass
      select2-rails (~> 4.0)
      xdan-datetimepicker-rails (~> 2.5.1)
    activejob (6.0.3.1)
      activesupport (= 6.0.3.1)
      globalid (>= 0.3.6)
    activemodel (6.0.3.1)
      activesupport (= 6.0.3.1)
    activerecord (6.0.3.1)
      activemodel (= 6.0.3.1)
      activesupport (= 6.0.3.1)
    activestorage (6.0.3.1)
      actionpack (= 6.0.3.1)
      activejob (= 6.0.3.1)
      activerecord (= 6.0.3.1)
      marcel (~> 0.3.1)
    activesupport (6.0.3.1)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
      zeitwerk (~> 2.2, >= 2.2.2)
    addressable (2.7.0)
      public_suffix (>= 2.0.2, < 5.0)
    airbrake (11.0.0)
      airbrake-ruby (~> 5.0)
    airbrake-ruby (5.0.2)
      rbtree3 (~> 0.5)
    arbre (1.2.1)
      activesupport (>= 3.0.0)
    aws-eventstream (1.1.0)
    aws-partitions (1.345.0)
    aws-sdk-core (3.104.3)
      aws-eventstream (~> 1, >= 1.0.2)
      aws-partitions (~> 1, >= 1.239.0)
      aws-sigv4 (~> 1.1)
      jmespath (~> 1.0)
    aws-sdk-kms (1.36.0)
      aws-sdk-core (~> 3, >= 3.99.0)
      aws-sigv4 (~> 1.1)
    aws-sdk-s3 (1.75.0)
      aws-sdk-core (~> 3, >= 3.104.1)
      aws-sdk-kms (~> 1)
      aws-sigv4 (~> 1.1)
    aws-sigv4 (1.2.1)
      aws-eventstream (~> 1, >= 1.0.2)
    bcrypt (3.1.13)
    bootsnap (1.4.6)
      msgpack (~> 1.0)
    builder (3.2.4)
    byebug (11.1.3)
    coderay (1.1.2)
    concurrent-ruby (1.1.6)
    content_disposition (1.0.0)
    crack (0.4.3)
      safe_yaml (~> 1.0.0)
    crass (1.0.6)
    devise (4.7.1)
      bcrypt (~> 3.0)
      orm_adapter (~> 0.1)
      railties (>= 4.1.0)
      responders
      warden (~> 1.2.3)
    diff-lcs (1.3)
    down (5.1.1)
      addressable (~> 2.5)
    erubi (1.9.0)
    factory_bot (5.2.0)
      activesupport (>= 4.2.0)
    factory_bot_rails (5.2.0)
      factory_bot (~> 5.2.0)
      railties (>= 4.2.0)
    faraday (1.0.1)
      multipart-post (>= 1.2, < 3)
    fast_jsonapi (1.5)
      activesupport (>= 4.2)
    ffi (1.12.2)
    formtastic (3.1.5)
      actionpack (>= 3.2.13)
    formtastic_i18n (0.6.0)
    globalid (0.4.2)
      activesupport (>= 4.2.0)
    has_scope (0.7.2)
      actionpack (>= 4.1)
      activesupport (>= 4.1)
    hashdiff (1.0.1)
    i18n (1.8.2)
      concurrent-ruby (~> 1.0)
    inherited_resources (1.11.0)
      actionpack (>= 5.0, < 6.1)
      has_scope (~> 0.6)
      railties (>= 5.0, < 6.1)
      responders (>= 2, < 4)
    jmespath (1.4.0)
    jquery-rails (4.4.0)
      rails-dom-testing (>= 1, < 3)
      railties (>= 4.2.0)
      thor (>= 0.14, < 2.0)
    jwt (2.2.1)
    kaminari (1.2.1)
      activesupport (>= 4.1.0)
      kaminari-actionview (= 1.2.1)
      kaminari-activerecord (= 1.2.1)
      kaminari-core (= 1.2.1)
    kaminari-actionview (1.2.1)
      actionview
      kaminari-core (= 1.2.1)
    kaminari-activerecord (1.2.1)
      activerecord
      kaminari-core (= 1.2.1)
    kaminari-core (1.2.1)
    listen (3.2.1)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    loofah (2.5.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    marcel (0.3.3)
      mimemagic (~> 0.3.2)
    method_source (1.0.0)
    mimemagic (0.3.5)
    mini_mime (1.0.2)
    mini_portile2 (2.4.0)
    minitest (5.14.1)
    msgpack (1.3.3)
    multipart-post (2.1.1)
    nio4r (2.5.2)
    nokogiri (1.10.9)
      mini_portile2 (~> 2.4.0)
    orm_adapter (0.5.0)
    pg (1.2.3)
    polyamorous (2.3.2)
      activerecord (>= 5.2.1)
    pry (0.13.1)
      coderay (~> 1.1)
      method_source (~> 1.0)
    public_suffix (4.0.5)
    puma (4.3.3)
      nio4r (~> 2.0)
    rack (2.2.2)
    rack-cors (1.1.1)
      rack (>= 2.0.0)
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails (6.0.3.1)
      actioncable (= 6.0.3.1)
      actionmailbox (= 6.0.3.1)
      actionmailer (= 6.0.3.1)
      actionpack (= 6.0.3.1)
      actiontext (= 6.0.3.1)
      actionview (= 6.0.3.1)
      activejob (= 6.0.3.1)
      activemodel (= 6.0.3.1)
      activerecord (= 6.0.3.1)
      activestorage (= 6.0.3.1)
      activesupport (= 6.0.3.1)
      bundler (>= 1.3.0)
      railties (= 6.0.3.1)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.3.0)
      loofah (~> 2.3)
    railties (6.0.3.1)
      actionpack (= 6.0.3.1)
      activesupport (= 6.0.3.1)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.20.3, < 2.0)
    rake (13.0.1)
    ransack (2.3.2)
      activerecord (>= 5.2.1)
      activesupport (>= 5.2.1)
      i18n
      polyamorous (= 2.3.2)
    rb-fsevent (0.10.4)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    rbtree3 (0.6.0)
    require_all (1.5.0)
    responders (3.0.1)
      actionpack (>= 5.0)
      railties (>= 5.0)
    rspec (3.9.0)
      rspec-core (~> 3.9.0)
      rspec-expectations (~> 3.9.0)
      rspec-mocks (~> 3.9.0)
    rspec-core (3.9.2)
      rspec-support (~> 3.9.3)
    rspec-expectations (3.9.2)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.9.0)
    rspec-mocks (3.9.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.9.0)
    rspec-rails (4.0.1)
      actionpack (>= 4.2)
      activesupport (>= 4.2)
      railties (>= 4.2)
      rspec-core (~> 3.9)
      rspec-expectations (~> 3.9)
      rspec-mocks (~> 3.9)
      rspec-support (~> 3.9)
    rspec-support (3.9.3)
    safe_yaml (1.0.5)
    sass (3.7.4)
      sass-listen (~> 4.0.0)
    sass-listen (4.0.0)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
    sassc (2.3.0)
      ffi (~> 1.9)
    sassc-rails (2.1.2)
      railties (>= 4.0.0)
      sassc (>= 2.0)
      sprockets (> 3.0)
      sprockets-rails
      tilt
    select2-rails (4.0.13)
    shrine (3.2.2)
      content_disposition (~> 1.0)
      down (~> 5.1)
    spring (2.1.0)
    spring-watcher-listen (2.0.1)
      listen (>= 2.7, < 4.0)
      spring (>= 1.2, < 3.0)
    sprockets (4.0.0)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    thor (1.0.1)
    thread_safe (0.3.6)
    tilt (2.0.10)
    tzinfo (1.2.7)
      thread_safe (~> 0.1)
    vcr (5.1.0)
    warden (1.2.8)
      rack (>= 2.0.6)
    webmock (3.8.3)
      addressable (>= 2.3.6)
      crack (>= 0.3.2)
      hashdiff (>= 0.4.0, < 2.0.0)
    websocket-driver (0.7.1)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.4)
    xdan-datetimepicker-rails (2.5.4)
      jquery-rails
      rails (>= 3.2.16)
    zeitwerk (2.3.0)

PLATFORMS
  ruby

DEPENDENCIES
  activeadmin
  activeadmin_addons
  airbrake
  aws-sdk-s3
  bootsnap (>= 1.4.2)
  byebug
  devise
  factory_bot_rails
  faraday
  fast_jsonapi
  jwt
  listen (~> 3.2)
  pg
  pry
  puma (~> 4.1)
  rack-cors
  rails (~> 6.0.3)
  rspec
  rspec-rails
  shrine
  spring
  spring-watcher-listen (~> 2.0.0)
  tzinfo-data
  vcr
  webmock

RUBY VERSION
   ruby 2.6.3p62

BUNDLED WITH
   1.17.3
```

**Comment by rjherrera** [comment]
@islador are you still experiencing this issue? I wasn't able to reproduce it either, but it has been a while!

**Comment by islador** [comment]
I worked around it as mentioned in the op email.

One thing of note that isn't in here; my rails 6.3.0.1 was built in API mode leading to missing dependencies for the Active Admin install which I manually placed. Perhaps that is the root here?

**Comment by rjherrera** [comment]
Oh ok. It could be, but I tried again with a clean installation using the `--api` flag, then installed Active Admin, with the work needed to make it work with the API mode, and afterwards ran `rails g activeadmin_addons:install` and it worked perfectly. Everything with the versions you used.

It could be down to the way you managed to make Active Admin work with the API mode, I followed [this article](https://medium.com/alturasoluciones/how-to-set-up-rails-api-app-to-use-activeadmin-79b418df8aad) and [this question](https://stackoverflow.com/questions/58339607/why-does-rails-fails-to-boot-with-expected-to-find-a-manifest-file-in-app-asse) to make it work, and after I tested that it was working seamlessly, I went and installed AA Addons, working without any issues.

I'm closing this issue as we can't seem to reproduce it and also, because you managed to make it work regardless. Thanks for your feedback though, we will keep an eye on it.
